### PR TITLE
fix: Fixed dependency issue that causes pip command to raise error

### LINKED ISSUE #664: Bug: pip package manager related commands raises error
**Describe the bug**

`pip freeze`, `pip install` and other various pip related commands doesn't work and raises error due to the `six` older version package.
It raises `ImportError: cannot import name 'ensure_str' from 'six'`

**To Reproduce**

1. If you have a virtualenv folder then delete it and recreate the virtual environment as specified in the [docs](https://github.com/anitab-org/mentorship-backend#run-app)
2. Activate the virtual environment
3. Run `pip install -r requirements.txt` command.
4. Now to see the error run any `pip` related commands. For example run `pip freeze` command.
5. See the error

**Expected behavior**
`pip` related commands must be executed without running into any error.

**Screenshots**
Screenshot from 2020-07-14 13-50-42

### PR DESCRIPTION
**Pull request by devkapilbansal**
### Description

Fixed issue when pip package manager related commands raise errors.

Fixes #664 

### Type of Change:

- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

- Run `python run.py`.
- Run `python -m unittest discover tests`

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Update requirements.txt

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by devkapilbansal** [comment]
@PrashanthPuneriya I removed some dependencies from requirements. Also, after researching I found that "ensure_str" was included in six==1.12.0. This issue may be due to old virtualenv. This issue was introduced in `virtualenv==20.0.0`.

I tested it with the same configuration provided by you.

**Review by SanketDG — COMMENTED** [review]
Looks great, I am pretty sure we can do a more extensive audit and remove many more dependencies.

**Comment by devkapilbansal** [comment]
Yes, @SanketDG in my opinion it's never a good choice to copy whole pip freeze to requirements file as some versions can cause issues later

**Review by PrashanthPuneriya — COMMENTED** [review]
Changing the requirements.txt gives following compatabiltiy issues:-
image
Though I get the compatability issues. It works fine :)

> I found that "ensure_str" was included in six==1.12.0.

But, we still do I have six==1.11.0 in the requirements.txt file right? Though this fixed the issue I still don't have any idea why it did. Can you please explain?

**Comment by devkapilbansal** [comment]
While installing we have to use url-resolver, that will be taken into effect from October, 2020. Also, there were some dependencies and we don't need to manage their versions. I removed those dependencies so that pip can install the most updated version automatically.
By letting pip choose the version for those dependencies, the issue got fixed.

**Comment by devkapilbansal** [comment]
@PrashanthPuneriya I will remove this conflicts soon.

**Comment by PrashanthPuneriya** [comment]
> While installing we have to use url-resolver, that will be taken into effect from October, 2020. Also, there were some dependencies and we don't need to manage their versions. I removed those dependencies so that pip can install the most updated version automatically.
> By letting pip choose the version for those dependencies, the issue probably got fixed.

Okay! But, what change has exactly solved this issue?
@SanketDG Is this PR valid since, it solves more than(to my knowledge) what the issue has stated?

**Comment by devkapilbansal** [comment]
This issue arises due to dependencies of awsebcli that were included in requirements. I removed those dependencies only.

**Comment by devkapilbansal** [comment]
@PrashanthPuneriya compatibility issues are solved now.

**Review by ravi5175 — DISMISSED** [review]

**Comment by devkapilbansal** [comment]
@SanketDG @PrashanthPuneriya can you please review it once?

File: README.md [hunk 1]
```diff
@@ -38,17 +38,20 @@ To setup the project locally read these wiki pages and follow the instructions:
 
 The project runs on Python 3.
 
-1. Create a virtual environment:
+1. Install latest version of virtualenv:
+`pip install -U virtualenv`
```
Comment (SanketDG) [inline README.md]: Can you use ```

So that it looks like

```
pip install -U virtualenv
```

**Review by SanketDG — DISMISSED** [review]
Looks good to me, I guess you could remove many more packages

Pinging @PrashanthPuneriya to see if this is okay to go ahead with

**Comment by devkapilbansal** [comment]
> Looks good to me, I guess you could remove many more packages
> 
> Pinging @PrashanthPuneriya to see if this is okay to go ahead with

@PrashanthPuneriya already open a new issue for that. If someone don't solve it then I will remove and update the dependencies. Here, I focus only on those that might create a problem for six

File: README.md (hunk 1 above)
Comment (PrashanthPuneriya) [reply README.md]: Formatting is fine @SanketDG. It follows the same style as steps no. 2, 3,....

**Review by PrashanthPuneriya — DISMISSED** [review]
LGTM! I have tested this locally and it works as expected. Haven't encountered any issues. We can go ahead with this.
With this PR we have fixed all the errors we used to encounter while setting up the project @SanketDG @isabelcosta :) 
Thanks a lot for your contribution @KapilBansal320 :)

File: README.md (hunk 1 above)
Comment (SanketDG) [reply README.md]: Oh apologies, I meant, please use code blocks in all places for commands.

Comment (devkapilbansal) [reply README.md]: This is not related to the pip issue. @SanketDG @PrashanthPuneriya  should I open another pull request for this or make the changes in this pull request only?

Comment (SanketDG) [reply README.md]: Please make the changes here since changes are already there in README file

Comment (devkapilbansal) [reply README.md]: @SanketDG changes done. Please review again

File: README.md [hunk 2]
```diff
@@ -38,17 +39,32 @@ To setup the project locally read these wiki pages and follow the instructions:
...
+virtualenv venv --python=python3
+```
 
-3. Install all the dependencies in `requirements.txt` file:
-`pip install -r requirements.txt`
+3. Activate the virtual environment:
+```
+source /venv/Scripts/activate
+```
+for Git Bash Users,
```
Comment (PrashanthPuneriya) [inline README.md]: If possible do something with this line. It looks odd to me.

Comment (devkapilbansal) [reply README.md]: Do something, do what?

**Comment by SanketDG** [comment]
Feel free to change the commit message and PR title to something more explanatory

We have a [Commit Message Style Guide](https://github.com/anitab-org/portal/wiki/Commit-Message-Style-Guide)

File: README.md (hunk 2 above)
Comment (PrashanthPuneriya) [reply README.md]: image
I think it's better if we reverse the flow of command and description shown and use capitals and colon symbol.

Comment (devkapilbansal) [reply README.md]: Done

**Comment by devkapilbansal** [comment]
@PrashanthPuneriya @SanketDG I made the necessary changes as asked.

**Review by PrashanthPuneriya — DISMISSED** [review]
LGTM!

File: README.md [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (SanketDG) [inline README.md]: No need for this.

**Review by SanketDG — DISMISSED** [review]
Approved!

Pinging @isabelcosta for the final set of eyes on this!

**Comment by isabelcosta** [comment]
@SanketDG are the README changes needed?

**Comment by devkapilbansal** [comment]
> @SanketDG are the README changes needed?

@isabelcosta I found it necessary because some people might have older versions of virtualenv that can create problems.

File: README.md (hunk 3 above)
Comment (devkapilbansal) [reply README.md]: Added by mistake. Removed now

**Review by SanketDG — DISMISSED** [review]

**Comment by devkapilbansal** [comment]
@isabelcosta Please review it

**Review by PrashanthPuneriya — DISMISSED** [review]

**Comment by SanketDG** [comment]
> are the README changes needed?

I honestly don't think so, given just suggesting `pip install -U virtualenv` does not work at all on most systems because of pip being installed as a system library. Please correct me if I am wrong.

**Comment by devkapilbansal** [comment]
> I honestly don't think so, given just suggesting `pip install -U virtualenv` does not work at all on most systems because of pip being installed as a system library. Please correct me if I am wrong.

@SanketDG the problem occurs in old versions of virtualenv. If we are proceeding by making virtualenv first (as explained in docs), then why `pip install -U virtualenv` will not work. Also, can you please explain what you mean by 
> pip being installed as a system library
If README changes still seems unnecessary, then I will revert them.

**Comment by SanketDG** [comment]
> Also, can you please explain what you mean by

That pip is installed generally as root, so if you are not the root user, you have to use `sudo`.

**Comment by devkapilbansal** [comment]
> That pip is installed generally as root, so if you are not the root user, you have to use `sudo`.
Then pip may not work for installing dependencies too. :thinking:

**Comment by SanketDG** [comment]
> Then pip may not work for installing dependencies too.

No, because then you'd be already inside the virtualenv, right?

**Comment by devkapilbansal** [comment]
Okay @SanketDG should I remove changes in README.md?? Although this is not only for Linux and they may use sudo if needed while updating virtualenv

**Comment by SanketDG** [comment]
I think the code block changes are good, just remove the installation step

**Comment by devkapilbansal** [comment]
> I think the code block changes are good, just remove the installation step

@SanketDG changes done now.

**Review by PrashanthPuneriya — APPROVED** [review]

**Review by vj-codes — APPROVED** [review]
LGTM 🎉

**Review by SanketDG — APPROVED** [review]
Looks good!

**Comment by devkapilbansal** [comment]
@PrashanthPuneriya can you please test and verify that the pip dependency issues are solved now. Once it is done we can move it to ready to merge

**Comment by PrashanthPuneriya** [comment]
Tested locally and haven't encountered any dependency issues or any other issues. It is safe to merge this PR.

**Review by isabelcosta — APPROVED** [review]
good work @devkapilbansal !
I deployed this to a Heroku link and this worked fine 🎉

**Pull request merged**